### PR TITLE
feat: add error handling for invalid offset

### PR DESCRIPTION
**Pull request by craycrayfish**
## Summary

Add error handling where we skip attempting to assign partition offsets if they are invalid.

## Motivation

Issues were observed where deployments could not assign partitions when the `get_watermark_offsets` method was returning invalid offset values.

## Author Checklist

- [x] Changes are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Documentation is updated if necessary.
- [x] Status checks pass.
- [ ] If the version number is changed, it is updated in `pyproject.toml` , `aineko/__init__.py`, `aineko/templates/first_aineko_pipeline/{{cookiecutter.project_slug}}/pyproject.toml`.

## Reviewer Checklist

- [ ] Title is accurate and formatted appropriately.
- [ ] No unnecessary changes are introduced.
- [ ] Description motivates each change.
- [ ] Docs are updated if necessary.
- [ ] Code is pythonic and consistent with the rest of the codebase.
- [ ] The code is consistent with the [style guide](https://google.github.io/styleguide/pyguide.html).

### CODE REVIEW

**Review by Barabazs — COMMENTED** [review]
Can you add test cases that validate this fix?

File: aineko/core/dataset.py [hunk 1]
```diff
@@ -153,12 +154,19 @@ def _update_offset_to_latest(self) -> None:
...
         for partition in partitions:
-            partition.offset = (
-                self.consumer.get_watermark_offsets(
-                    partition, cached=self.cached
-                )[1]
-                - 1
-            )
+            high_offset = self.consumer.get_watermark_offsets(
+                partition, cached=self.cached
+            )[1]
```
Comment (giddyphysicist) [inline aineko/core/dataset.py:159]: is there any chance this line could fail? for example, if get_watermark_offsets didn't return a tuple, then the [1] would error. if the method is strict in what it can return, then it should be fine

Comment (mdigi14) [reply aineko/core/dataset.py:159]: We should handle this upstream in `_update_offset_to_latest`. If this function fails because `get_watermark_offsets()` returns a `KafkaException` then I think we could either gracefully fail by using the same offset as the last call or raise a fatal error.

**Comment by craycrayfish** [comment]
> Can you add test cases that validate this fix?

Added a test to check that these methods work when calling on an empty dataset, which was an edge case we suspected was causing this issue. Admittedly it doesn't cover the full spectrum of possible kafka scenarios.

File: aineko/core/dataset.py (hunk 1 above)
Comment (craycrayfish) [reply aineko/core/dataset.py:159]: Added an error case where the message defaults to the next message if offset cannot be updated.

File: tests/integration/test_kafka_edge_cases.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    except ray.exceptions.RayActorError:
+        consumer = DatasetConsumer(
+            dataset_name="test_result",
+            node_name="consumer",
+            pipeline_name="integration_test_kafka_edge_cases",
+            dataset_config={},
+            has_pipeline_prefix=True,
+        )
+        count_messages = consumer.consume_all(end_message="END")
+        assert count_messages[0]["message"] == "OK"
```
Comment (Barabazs) [inline tests/integration/test_kafka_edge_cases.py]: At first glance I was confused by what you're doing here. I don't think it's very common to use a try-except like that. I would suggest refactoring it so it's easier to understand to everyone.

```suggestion
    try:
        runner.run()
    except ray.exceptions.RayActorError:
        # This is expected because we activated the poison pill
        pass

    consumer = DatasetConsumer(
        dataset_name="test_result",
        node_name="consumer",
        pipeline_name="integration_test_kafka_edge_cases",
        dataset_config={},
        has_pipeline_prefix=True,
    )
    count_messages = consumer.consume_all(end_message="END")
    assert count_messages[0]["message"] == "OK"
```

Comment (Barabazs) [reply tests/integration/test_kafka_edge_cases.py]: Also fwiw this test passes on the current develop, so without your fix. I assume this is because the dataset wasn't purged locally since I first ran the test on this branch and then on develop. If that's the case, our tests would need to do some clean-up before and after running. What do you think?

**Review by Barabazs — APPROVED** [review]
I left a non-blocking suggestion.

File: tests/integration/test_kafka_edge_cases.py (hunk 2 above)
Comment (mdigi14) [reply tests/integration/test_kafka_edge_cases.py]: I agree with Erwin on not putting logic inside an `except` block. It should only be used for error handling.

**Review by mdigi14 — APPROVED** [review]
